sbt-test/scala2-compat/erasure/scala2Lib/Api.scala

```diff
@@ -186,4 +186,10 @@ class Z {
   def objectARRAY_88(x: Array[Any]): Unit = {}
   def objectARRAY_89(x: Array[AnyRef]): Unit = {}
   def objectARRAY_90(x: Array[AnyVal]): Unit = {}
+
+  def objectARRAY_91(x: Array[Nothing]): Unit = {}
+  def objectARRAY_92(x: Array[Null]): Unit = {}
+  def objectARRAY_93(x: Array[_ <: Nothing]): Unit = {}
+  def objectARRAY_94(x: Array[_ <: Null]): Unit = {}
```

Comment:
This should actually be:
```suggestion
  def nothing$ARRAY_93(x: Array[_ <: Nothing]): Unit = {}
  def null$ARRAY_94(x: Array[_ <: Null]): Unit = {}
```
but for some reason, it passes the test locally, I will have to test it better, but my machine keeps crashing... so checking with the CI :)

Reply:
So this is weird, the CI passes while `Array[_ <: Nothing]` erasure is wrong. I've tested with the following:

```scala
// In Scala 2
class Foo {
    def a: Array[Nothing] = null
    def b: Array[Null] = null
    def c: Array[_ <: Nothing] = null
    def s: Array[_ <: Null] = null
}
```


```scala
// In Scala 3
@main def Test =
    (new Foo).a
    (new Foo).b
    (new Foo).c
    (new Foo).s
```

The generated bytecode from Scala 2 is the following:
```java
Compiled from "s2.scala"
public class Foo {
  public java.lang.Object[] a();
  public java.lang.Object[] b();
  public scala.runtime.Nothing$[] c();
  public scala.runtime.Null$[] s();
  public Foo();
}
```

We can clearly see Scala 3 (this PR) emitting a call to the wrong method for `Foo.c`:
```
public final class s3$package$ implements java.io.Serializable {
  public static final s3$package$ MODULE$;

  public static {};
    Code:
       0: new           #2                  // class s3$package$
       3: dup
       4: invokespecial #16                 // Method "<init>":()V
       7: putstatic     #18                 // Field MODULE$:Ls3$package$;
      10: return

  public scala.runtime.Null$[] Test();
    Code:
       0: new           #29                 // class Foo
       3: dup
       4: invokespecial #30                 // Method Foo."<init>":()V
       7: invokevirtual #34                 // Method Foo.a:()[Ljava/lang/Object;
      10: pop
      11: new           #29                 // class Foo
      14: dup
      15: invokespecial #30                 // Method Foo."<init>":()V
      18: invokevirtual #37                 // Method Foo.b:()[Ljava/lang/Object;
      21: pop
      22: new           #29                 // class Foo
      25: dup
      26: invokespecial #30                 // Method Foo."<init>":()V
      29: invokevirtual #40                 // Method Foo.c:()Ljava/lang/Object;
      32: pop
      33: new           #29                 // class Foo
      36: dup
      37: invokespecial #30                 // Method Foo."<init>":()V
      40: invokevirtual #43                 // Method Foo.s:()[Lscala/runtime/Null$;
      43: areturn
}
```

Reply:
I've checked a bit and throwing a `LinkageError` will not be considered a failure... but the problem is that there is a linkage error before the code I've added (I still need to find which one).

Reply:
There is already an issue opened on sbt (https://github.com/sbt/sbt/issues/6852). The proposed workaround (`fork := true`) worked locally